pkg/etcd/etcd.go

```diff
@@ -1204,6 +1256,28 @@ func (e *ETCD) setEtcdStatusCondition(node *v1.Node, client corev1.CoreV1Interfa
 	return nodeHelper.PatchNodeStatus(client, types.NodeName(node.Name), node, &updatedNode)
 }
 
+// Search for the ready condition and check if it is true. If it is not true, we should maintain the etcd status condition
+// as false, so that we don't have a false positive on the etcd status condition.
+func (e *ETCD) setEtcdStatusConditionIfNodeIsNotReady(node *v1.Node, client corev1.CoreV1Interface) (*v1.Node, []byte, error) {
+	updatedNode := *node
+	if find, readyCondition := nodeUtil.GetNodeCondition(&updatedNode.Status, readyStatusType); find >= 0 {
```

Comment:
I don't think the ask was to set the etcd condition based on the kubelet ready status. The etcd condition should be based on whether or not the node is a member of the etcd cluster. If the the etcd member list does not contain this member, THEN we should change the status as shown below.

Similarly, if the member is in the cluster but unhealthy as per the etcd member status, then we should set the etcd condition accordingly.  You might look at the etcd member status and health outputs, as described here:
https://etcd.io/docs/v3.5/tutorials/how-to-check-cluster-status/